src/hsd/parser.py

```diff
@@ -286,8 +286,8 @@ def _starttag(self, tagname, closeprev):
         if len(tagname_stripped.split()) > 1:
             self._error(SYNTAX_ERROR, (self._currline, self._currline))
         self._hsdoptions[common.HSDATTR_LINE] = self._currline
-        self._hsdoptions[common.HSDATTR_TAG] = tagname_stripped
-        tagname_stripped = tagname_stripped.lower()
+        #self._hsdoptions[common.HSDATTR_TAG] = tagname_stripped
+        #tagname_stripped = tagname_stripped.lower()
```

Comment:
This change does make the HSD parser case sensitive which is not really in line with the reference implementation.
From the DFTB+ manual:
> The name of the properties, the methods and the logical values are case insensitive

Reply:
That's true. However, IMO it is better, if the parser just passes the tokens literally. The tree builder can then decide to convert everything to lower case (and eventually store the original form somewhere for later reference...)

Reply:
I totally agree, forcing a case-insensitive input will limit the applicability of HSD. But with this change there should be an easy way or a recipe to generate a DFTB+ compatible parser.

Than again, one has also to think about clarifying this part about HSD in the DFTB+ manual, which is the only existing specification.